### PR TITLE
fix(span-tree): Horizontal autoscroll improvements

### PR DESCRIPTION
**Pull request by 0Calories**
This PR adds the following:

- Fixes a bug where spans without any descendants were not getting considered when determining horizontal positioning
- Horizontal positioning has been changed so that the entire child span toggle button remains in view, instead of half
- Fixes a bug where horizontal positioning was inconsistent when visiting the event view from an anchor link

### CODE REVIEW

File: static/app/components/events/interfaces/spans/scrollbarManager.tsx [hunk 1]
```diff
@@ -84,6 +87,26 @@ export class Provider extends Component<Props, State> {
...
+    if (anchoredSpanHash) {
+      // We cannot assume the root is in view to start off, if there is an anchored span
+      this.spansInView.isRootSpanInView = false;
+      this.spanBars.forEach(spanBar => spanBar.disconnectObservers());
+
+      setTimeout(() => {
+        this.spanBars.forEach(spanBar => spanBar.connectObservers());
+      }, 500);
+      return;
+    }
```
Comment (k-fish) [inline static/app/components/events/interfaces/spans/scrollbarManager.tsx:107]: Setting a fixed timeout will be brittle, so I'm not sure about this. One alternative off the top of my head is doing an interval that checks for the existence of the anchored span in the viewport and then cancels itself instead to make it at least re-connect post-scroll consistently vs. randomly.

**Review by k-fish — APPROVED** [review]
I'm not sure about that turning on/off observers with a fixed timeout, but looks functional and otherwise looks fine.

File: static/app/components/events/interfaces/spans/scrollbarManager.tsx (hunk 1 above)
Comment (0Calories) [reply static/app/components/events/interfaces/spans/scrollbarManager.tsx:107]: @k-fish That's a great idea! I'll look into setting this up tomorrow

File: static/app/components/events/interfaces/spans/scrollbarManager.tsx [hunk 2]
```diff
@@ -84,6 +87,42 @@ export class Provider extends Component<Props, State> {
...
+        }
+      }, 100);
+
+      // If the anchored span is never found in the view (malformed ID), cancel the interval
+      setTimeout(() => {
+        if (this.anchorCheckInterval) {
+          clearInterval(this.anchorCheckInterval);
+          this.anchorCheckInterval = null;
+        }
+      }, 5000);
```
Comment (k-fish) [inline static/app/components/events/interfaces/spans/scrollbarManager.tsx:120]: I'd switch this to `1000` or even `500`

**Review by k-fish — APPROVED** [review]

**Pull request merged**